app/models/job/Job.scala

```diff
@@ -286,6 +287,18 @@ class JobService @Inject()(wkConf: WkConf,
     ()
   }
 
+  def cleanUpIfFailed(job: Job): Fox[Unit] =
+    if (job.state == JobState.FAILURE && job.command == "convert_to_wkw") {
+      logger.info(s"Job ${job._id} failed. Deleting Dataset...")
+      val commandArgs = job.commandArgs.value
+      for {
+        datasetName <- commandArgs.get("dataset_name").map(_.as[String]).toFox
+        organizationName <- commandArgs.get("organization_name").map(_.as[String]).toFox
+        dataset <- dataSetDAO.findOneByNameAndOrganizationName(datasetName, organizationName)(GlobalAccessContext)
+        _ <- dataSetDAO.deleteDataset(dataset._id)
```

Comment:
After a failed job, what exactly is removed?
 - dataset entry from postgres (should definitely be removed, I think that is what you do here already)
 - is anything deleted on the datastore side?
   - I think the target directory of the dataset (if it has already been created by the conversion job) should be deleted as well
   - I’d say the directory in `.converting` should be kept
   - I’d say the directory in `.forConversion` should also be kept
   - those two have the job id or some timestamp in their path anyway, right? :thinking: so that a retry with the same name will not cause problems

Reply:
- The dataset entry is removed from the progres.
- I think the target directory cannot exist if the job fails, because the directory in `.converting`  gets moved to the target directory after a successful conversion.
- The directory in `.converting`  is deduplicated by the request id.
- The directory in `.forConversion` is not deduplicated, so it would need to be deleted, or renamed. Do you have any preferences on this?

Reply:
Thanks for the explanations! You are right, the target directory should not exist yet, so that’ll be fine.

I think it would be cool to deduplicate the contents of .forConversion somehow. Possibly with the upload id /timestamp? :thinking: will have to be passed to the worker job somehow, though. I am not sure how this works currently. If you can find an easy solution there, feel free to implement it, otherwise we can talk about this again :)

Reply:
I have implemented the deduplication step in the worker, which should do the trick :slightly_smiling_face: